executor/predictor/metadata.go

```diff
@@ -0,0 +1,82 @@
+package predictor
+
+import (
+	"github.com/seldonio/seldon-core/operator/apis/machinelearning.seldon.io/v1"
+	"sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+type MetadataTensor struct {
+	DataType string `json:"datatype,omitempty"`
+	Name     string `json:"name,omitempty"`
+	Shape    []int  `json:"shape,omitempty"`
+}
+
+type ModelMetadata struct {
+	Name     string           `json:"name,omitempty"`
+	Platform string           `json:"platform,omitempty"`
+	Versions []string         `json:"versions,omitempty"`
+	Inputs   []MetadataTensor `json:"inputs,omitempty"`
+	Outputs  []MetadataTensor `json:"outputs,omitempty"`
+}
+
+type GraphMetadata struct {
+	Name         string                   `json:"name"`
+	Models       map[string]ModelMetadata `json:"models"`
+	GraphInputs  []MetadataTensor         `json:"graphinputs"`
+	GraphOutputs []MetadataTensor         `json:"graphoutputs"`
+}
+
+func NewGraphMetadata(p *PredictorProcess, spec *v1.PredictorSpec) (*GraphMetadata, error) {
+	models, err := p.MetadataMap(spec.Graph)
+	if err != nil {
+		return nil, err
+	}
+	output := &GraphMetadata{
+		Name: spec.Name,
+		Models: models,
+	}
+	output.GraphInputs, output.GraphOutputs = output.GetShapeFromGraph(spec.Graph)
+	return output, nil
+}
+
+func (gm *GraphMetadata) GetShapeFromGraph(node *v1.PredictiveUnit) (
```

Comment:
Is this graph expected to be called from other packages? if not best ot keep it lower case

Reply:
No, it's not. Will move to lower case